profiles/common/modules/custom/tmgmt_workbench/tmgmt_workbench.module

```diff
@@ -114,74 +116,91 @@ function tmgmt_workbench_form_tmgmt_entity_ui_translate_form_alter(&$form, $form
       $state = $node->workbench_moderation['current']->state;
       $translation_possible = in_array($state, $states, 'strict');
 
-      $languages = tmgmt_available_languages();
-      $options = &$form['languages']['#options'];
+      $options = $form['languages']['#options'];
+
+      $existing_jobs = tmgmt_job_item_load_latest(
+        'workbench_moderation',
+        $entity_type,
+        $node->nid,
+        $node->translations->original
+      );
 
-      $existing_jobs = tmgmt_job_item_load_latest('workbench_moderation',
-        $entity_type, $node->nid, $node->translations->original);
       $always_show_direct_translation_links = variable_get('tmgmt_workbench_moderation_always_show_direct_translation_links', FALSE);
 
-      // Disable checkbox for languages already translated.
-      foreach ($languages as $langcode => $language) {
-        if (isset($node->translations->data[$langcode]) && !empty($existing_jobs)) {
-          $form['languages'][$langcode] = array(
+      foreach ($options as $lang_code => $option) {
+        // Disable checkbox for languages already translated.
+        if (isset($node->translations->data[$lang_code]) && !empty($existing_jobs)) {
```

Comment:
IMO, the code would be more efficient if `$node->translations->data` and and `$node->translations->original` are stored in 2 variables before the loop and then, the loop treatment will be done on them. That prevents the process to browse each time, the node object and the translation objects before to get required data.